var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -210,19 +216,20 @@ def install(self, spec, prefix):
         #
         # Below we try to match the entire HDF5_LIB line and substitute
         # with the list of libraries that needs to be linked.
-        if self.spec.variants['hdf5'].value != 'none':
+        if spec.variants['hdf5'].value != 'none':
             make_inc = join_path(self.stage.source_path, 'make.inc')
             hdf5_libs = ' '.join(spec['hdf5:hl,fortran'].libs)
             filter_file(r'HDF5_LIB([\s]*)=([\s\w\-\/.,]*)',
                         'HDF5_LIB = {0}'.format(hdf5_libs),
                         make_inc)
-            if self.spec.variants['hdf5'].value == 'serial':
-                # Note that there is a benign side effect with this filter
-                # file statement. It replaces an instance of MANUAL_DFLAGS
-                # that is a comment in make.inc.
-                filter_file(r'MANUAL_DFLAGS([\s]*)=([\s]*)',
-                            'MANUAL_DFLAGS = -D__HDF5_SERIAL',
-                            make_inc)
+            if spec.variants['hdf5'].value == 'serial':
+                if '+mpi' or '@develop' in spec:
```

Comment:
This will always evaluate to `True`, I think you mean something like

`if '+mpi' in spec or '@develop' in spec:`

However, given the above constraint, `hdf5=serial` is only allowed for `+mpi` or `@develop`, so I don't think this check is required.